pulsar-broker/src/test/java/org/apache/pulsar/discovery/service/web/DiscoveryServiceWebTest.java

```diff
@@ -56,6 +56,9 @@
 public class DiscoveryServiceWebTest extends ProducerConsumerBase {
 
     private final Client client = ClientBuilder.newClient(new ClientConfig().register(LoggingFeature.class));
+    // DiscoveryServiceServlet gets initialized by a server and this map will help to retrieve ZK while mocking
+    // DiscoveryServiceServlet
+    private static final Map<String, MetadataStoreExtended> metadataStoreInstanceCache = Maps.newConcurrentMap();
```

Comment:
it looks like we are not clearing this map at the end of the test.
we will leak MetadataStoreExtended instances